zebra-chain/src/orchard/tree.rs

```diff
@@ -498,8 +492,9 @@ impl From<&NoteCommitmentTree> for SerializedTree {
         // Convert the note commitment tree from
         // [`Frontier`](bridgetree::Frontier) to
         // [`CommitmentTree`](merkle_tree::CommitmentTree).
-        let tree = CommitmentTree::from_frontier(&tree.inner);
-        tree.write(&mut serialized_tree)
+        let tree = incrementalmerkletree::frontier::CommitmentTree::from_frontier(&tree.inner);
+
+        write_commitment_tree(tree, &mut serialized_tree)
```

Comment:
This looks like we could just call the function from `zcash_primitives`?

```suggestion
         zcash_primitives::merkle_tree::write_commitment_tree(tree, &mut serialized_tree)
```

Reply:
You are right, thanks. Done at https://github.com/ZcashFoundation/zebra/pull/7053/commits/ed57b42254dc6b2547cb6ccada2e6c332a0e575a